pkg/controllers/common/external_oidc.go

```diff
@@ -56,9 +56,17 @@ func AuthConfigCheckerInformers[T factory.Informer](c *AuthConfigChecker) []T {
 // that includes the structured auth-config ConfigMap, and the KAS args include the respective
 // arg that enables usage of the structured auth-config. It returns false otherwise.
 func (c *AuthConfigChecker) OIDCAvailable() (bool, error) {
-	if auth, err := c.authLister.Get("cluster"); err != nil {
-		return false, fmt.Errorf("getting authentications.config.openshift.io/cluster: %v", err)
-	} else if auth.Spec.Type != configv1.AuthenticationTypeOIDC {
+	var auth *configv1.Authentication
+	// Retry with exponential backoff to handle transient NotFound during upgrades.
+	err := retry.OnError(retry.DefaultBackoff, errors.IsNotFound, func() error {
```

Comment:
We can use our custom Backoff configurations, if we think that total ~1.6 seconds is insufficient.